packages/lesswrong/components/editor/FriendlyPostVersionHistory.tsx

```diff
@@ -0,0 +1,298 @@
+
+  const { document: revision, loading: revisionLoading } = useSingle({
+    skip: !selectedRevisionId,
+    documentId: selectedRevisionId||"",
+    collectionName: "Revisions",
+    fetchPolicy: "cache-first",
+    fragmentName: "RevisionDisplay",
+  });
+
+  const isLive = (r: {_id: string}) => r._id === post.contents_latest
+
+  return (
+    <LWDialog open={true} maxWidth={false} onClose={onClose}>
+      <div className={classes.root}>
+        <div className={classes.leftColumn}>
+          {revisions &&
+            revisions.map((rev) => (
+              <div
+                key={rev._id}
+                className={classNames(classes.revisionRow, {
+                  [classes.selectedRevision]: rev._id === selectedRevisionId,
+                })}
+                onClick={() => setSelectedRevisionId(rev._id)}
+              >
+                <LWTooltip title={isLive(rev) ? LIVE_REVISION_TOOLTIP : undefined} placement="top" popperClassName={classes.tooltip}>
+                  <span
+                    className={classNames(classes.versionNumber, {
+                      [classes.liveRevision]: isLive(rev),
+                    })}
+                  >
+                    {rev.version}
+                  </span>
+                </LWTooltip>
+                <ChangeMetricsDisplay changeMetrics={rev.changeMetrics} />
+                <span className={classes.editedAt}>
+                  <FormatDate date={rev.editedAt} />
+                </span>
+              </div>
+            ))}
+          <div className={classes.loadMore}>
+            <LoadMore {...loadMoreProps} />
+          </div>
+        </div>
+        <div className={classes.selectedRevisionDisplay}>
+          {revisionLoading && <Loading />}
+          {revision && (
+            <>
+              {canRevert && (
+                <div className={classes.header}>
+                  <div className={classes.titleRow}>
+                    <div className={classes.versionTitle}>
+                      v{revision.version}
+                      {isLive(revision) ? " (Live version)" : ""}
+                    </div>
+                    {!isCollabEditor && <LWTooltip title={LOAD_VERSION_TOOLTIP} placement="top"  popperClassName={classes.tooltip}>
+                      <EAButton
+                        variant="outlined"
+                        className={classes.button}
+                        onClick={() => loadVersion(revision.version)}
+                      >
+                        Load into editor
+                      </EAButton>
+                    </LWTooltip>}
+                    <LWTooltip title={RESTORE_VERSION_TOOLTIP} placement="top" popperClassName={classes.tooltip}>
+                      <EAButton variant="contained" className={classes.button} onClick={restoreVersion}>
+                        Restore {revertInProgress && <Loading />}
```

Comment:
nbd but this looks a bit weird, maybe better to replace the text with the loading component instead?

<img width="322" alt="Screen Shot 2024-02-20 at 7 41 07 PM" src="https://github.com/ForumMagnum/ForumMagnum/assets/9057804/cf3452d0-7c1c-4b96-ab41-bb6b2251eed5">